src/coord/rmr/conn.c

```diff
@@ -425,11 +426,31 @@ static int MRConn_SendAuth(MRConn *conn) {
   CONN_LOG(conn, "Authenticating...");
 
   // if we failed to send the auth command, start a reconnect loop
-  if (redisAsyncCommand(conn->conn, MRConn_AuthCallback, conn, "AUTH %s %s", clusterConfig.aclUsername, conn->ep.password) ==
-      REDIS_ERR) {
-    MRConn_SwitchState(conn, MRConn_ReAuth);
-    return REDIS_ERR;
+  size_t len = 0;
+
+  if (!IsEnterprise()) {
+    // Take the GIL before calling the internal function getter
+    RedisModule_ThreadSafeContextLock(RSDummyContext);
+    const char *internal_secret = RedisModule_GetInternalSecret(RSDummyContext, &len);
+    RedisModule_ThreadSafeContextUnlock(RSDummyContext);
```

Comment:
You should not release the GIL up until you finish using the `internal_secret`, otherwise that main thread might change it while you are using it.

Reply:
Right, fixed 👍